src/dotnet/APIView/APIViewWeb/Repositories/DevopsArtifactRepository.cs

```diff
@@ -83,13 +91,38 @@ private string GetArtifactRestAPIForRepo(string repoName)
 
         public async Task RunPipeline(string pipelineName, string reviewDetails, string originalStorageUrl)
         {
-            SetDevopsClientHeaders();
             //Create dictionary of all required parametes to run tools - generate-<language>-apireview pipeline in azure devops
             var reviewDetailsDict = new Dictionary<string, string> { { "Reviews", reviewDetails }, { "APIViewUrl", _hostUrl }, { "StorageContainerUrl", originalStorageUrl } };
-            var pipelineParams = new Dictionary<string, Dictionary<string, string>> { { "templateParameters", reviewDetailsDict } };
-            var stringContent = new StringContent(JsonSerializer.Serialize(pipelineParams), Encoding.UTF8, "application/json");
-            var response = await _devopsClient.PostAsync(_pipeline_run_rest, stringContent);
-            response.EnsureSuccessStatusCode();
+            var devOpsCreds = new VssBasicCredential("nobody", _configuration["Azure-Devops-PAT"]);
+            var devOpsConnection = new VssConnection(new Uri($"https://dev.azure.com/azure-sdk/"), devOpsCreds);
+            string projectName = _configuration["Azure-Devops-internal-project"] ?? "internal";
+
+            BuildHttpClient buildClient = await devOpsConnection.GetClientAsync<BuildHttpClient>();
+            var projectClient = await devOpsConnection.GetClientAsync<ProjectHttpClient>();
+            int definitionId = await GetPipelineId(pipelineName, buildClient, projectName);
+            if (definitionId == 0)
+            {
+                throw new Exception(string.Format("Azure Devops pipeline is not found in internal project with name {0}. Please recheck and ensure pipeline exists with this name", pipelineName));

```

Comment:
You mention internal explicitly should this use projectName?

Reply:
Correct. I will update it to projectName